### PR TITLE
[master]support create swap device with vdisk

### PR DESCRIPTION
**Pull request by trenxiaof**
https://github.ibm.com/zvc/planning/issues/10607

previous config:
swap_force_mdisk: False only works for swap only case( disk_pool is None, bfv + swap)

new config:
swap_default_with_mdisk: True   applies for all the cases when disk_pool is not None (including bfv+swap), if True, create swap with mdisk, if False, create swap device with vdisk

### CODE REVIEW

**Comment by trenxiaof** [comment]
截屏2024-03-19 18 06 03


@jichenjc @SeanHQF Did above tests, except for swap only case(it hungs for the issue:https://github.ibm.com/zvc/planning/issues/17425), all the other cases are as expected(keep the current default behavior and can controled by the new config swap_default_with_mdisk)

will continue to test swap only case.

**Comment by jichenjc** [comment]
@trenxiaof this is great, can you help fix the failed test then we can merge this

**Comment by trenxiaof** [comment]
please hold on the merge!!

File: zvmsdk/config.py [hunk 1]
```diff
@@ -716,6 +728,9 @@ def _check_value(self, conf):
                 # Convert type
                 if v2['type'] == 'int':
                     v2['default'] = int(v2['default'])
+                if v2['type'] == 'bool':
```
Comment (trenxiaof) [inline zvmsdk/config.py:731]: current zcc config file parse doesn't cover the bool type, the default config type is str, but the default value for the bool type is True or False, so the problem is if use the default config value, it's bool variable, if use the configured value, it's str type.  So change the code here to support the config with bool type.

**Comment by trenxiaof** [comment]
test the swap only case on  non-ssi host， all the results are OK
 On ssi host, sometime the deployement is OK, sometimes failed for the placement update problem
should be the same reason as https://github.ibm.com/zvc/planning/issues

so we can merge the pr first. Thanks @jichenjc  @SeanHQF

**Pull request merged**